src/main/java/com/nitnelave/CreeperHeal/block/CreeperBed.java

```diff
@@ -1,90 +1,62 @@
 package com.nitnelave.CreeperHeal.block;
 
-import org.bukkit.Material;
-import org.bukkit.block.Block;
-import org.bukkit.block.BlockFace;
 import org.bukkit.block.BlockState;
-import org.bukkit.material.Bed;
+import org.bukkit.block.data.BlockData;
+import org.bukkit.block.data.type.Bed;
 
-import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
 /**
- * Bed implementation of CreeperBlock.
- * 
- * @author nitnelave
- * 
+ * Bed implementation of CreeperMultiblock.
+ *
+ * @author Jikoo
  */
-class CreeperBed extends CreeperBlock
+class CreeperBed extends CreeperMultiblock
 {
 
-    /*
-     * The direction the bed is facing.
-     */
-    private final BlockFace orientation;
-
-    /*
-     * Constructor.
-     */
     CreeperBed(BlockState blockState)
     {
-        Bed bedData = castData(blockState, Bed.class);
-        orientation = bedData.getFacing();
-        Block block = blockState.getBlock();
-        if (!bedData.isHeadOfBed())
-            block = block.getRelative(orientation);
-        this.blockState = block.getState();
-    }
-
-    /**
-     * The blockstate is always the head of the bed, this gets the foot.
-     *
-     * @return the foot of the bed.
-     */
-    public Block getFoot()
-    {
-        return getBlock().getRelative(orientation.getOppositeFace());
+        super(blockState);
+
+        BlockData blockData = blockState.getBlockData();
+        if (!(blockData instanceof Bed))
+            throw new IllegalArgumentException("Invalid BlockData: " + blockData.getClass().getName());
+
+        Bed bed = ((Bed) blockData);
+        BlockState head, foot;
+
+        if (bed.getPart() == Bed.Part.HEAD)
+        {
+            head = blockState;
+            foot = blockState.getBlock().getRelative(bed.getFacing().getOppositeFace()).getState();
+        }
+        else
+        {
+            head = blockState.getBlock().getRelative(bed.getFacing()).getState();
+            foot = blockState;
+        }
+
+        // Ensure materials match.
+        if (head.getType() != foot.getType())
+            return;
```

Comment:
If they don't match, what do you do? You leave the object half-initialized? If it's an unrecoverable failure that is not supposed to happen, throw an exception, otherwise handle that case anyway.

Reply:
No, I opted not to add neighbors that don't exist. If the bed is malformed, we consider it half-by-half and restore it individually.